### ISSUE TITLE
Add a toggle to reveal the marks of evil.

### ISSUE THREAD

**Issue by gilch** [syl20bnr/spacemacs#5619]
Marks are not used as much in Vim as vi, thanks to visual mode and text objects, but they are still useful and Evil also has them. Evil's marks are invisible like vi's are. The only way to tell where they are is with `:marks`. Vim has several plugins to help with this. Evil also has one: the [evil-visual-mark-mode](https://github.com/roman/evil-visual-mark-mode) package, which shows them in-line as an overlay.

It would also be nice to briefly turn this on after hitting `m`, ``` ` ```, or `'`, but before selecting which character. This is consistent with the way Spacemacs shows other overlays only when needed (e.g. `:s` shows what will change as an overlay even before you do it).

The package is on MELPA, so it's possible now to add the package to the dotfile and add a custom toggle under `SPC o`, but it seems like something Spacemacs should include by default. I'm also not sure how to make the second part work.

**Comment by joehillen** [comment]
evil-visual-mark-mode seems reasonable to add to [spacemacs-packages](https://github.com/syl20bnr/spacemacs/blob/master/layers/%2Bdistribution/spacemacs/packages.el#L12) and on by default.

I'd suggest putting the toggle under `SPC t`, ```SPC t ` ``` seems like a natural choice.

**Comment by gilch** [comment]
```elisp
  (defun evil-visual-mark-around (orig-fun &rest args)
    (evil-visual-mark-show)
    (call-interactively orig-fun)
    (evil-visual-mark-hide))
  (advice-add 'evil-set-marker :around #'evil-visual-mark-around)
  (advice-add 'evil-goto-mark :around #'evil-visual-mark-around)
```
My first attempt at an as-needed overlay doesn't quite work right. You have to type the mark character twice, and if you interrupt with `C-g`, it doesn't turn off the overlay. Is this possible in elisp?

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
